torch/_dynamo/variables/misc.py

```diff
@@ -99,22 +103,28 @@ def _resolved_getattr_and_source(self, tx: "InstructionTranslator", name):
             type_to_use_source = self.objvar.source
 
         source = None
-        if self.objvar.source is not None:
-            # Walk the mro tuple to find out the actual class where the
-            # attribute resides.
-            search_mro = type_to_use.__mro__
-            start_index = search_mro.index(search_type) + 1
-            for index in range(start_index, len(search_mro)):
-                if hasattr(search_mro[index], name):
+        resolved_class = None
+        resolved_attr = None
+        search_mro = type_to_use.__mro__
+        start_index = search_mro.index(search_type) + 1
+        # Implemented based on https://github.com/python/cpython/blob/3.11/Objects/typeobject.c#L8812
+        # super has its getattro implementation. The key point is that instead of calling getattr, it checks the
+        # attribute in the class __dict__
+        for index in range(start_index, len(search_mro)):
+            # Dont call getattr, just check the __dict__ of the class
+            if resolved_getattr := search_mro[index].__dict__.get(name, NO_SUCH_SUBOBJ):
```

Comment:
This is the main change for the reland. Earlier this was `inspect.getattr_static`, which did not limit the search to just the class in question. Based on the CPython implementation, we can just search the __dict__ of the class.